src/main/java/io/github/mzmine/modules/io/import_rawdata_mzml/msdk/data/BuildingMzMLMsScan.java

```diff
@@ -747,24 +578,55 @@ public Optional<Integer> getScanNumber(String spectrumId) {
     return Optional.ofNullable(null);
   }
 
-  public void setAdvancedParameters(ParameterSet advancedParam) {
-    this.advancedParameters = advancedParam;
-  }
+  /**
+   * Called when spectrum end is read. Load, process data points and memory map resulting data to
+   * disk to save RAM.
+   */
+  public boolean loadProcessMemMapData(final MemoryMapStorage storage,
+      final MsProcessorList spectralProcessor) {
+    try {
+      var specData = loadData();
+      if (specData == null) {
+        // may be null for UV spectra
+        return false;
+      }
+
+      // process and filter - needs metadata so wrap
+      specData = spectralProcessor.processScan(this, specData);
+
+      // memory map
+      this.mzValues = StorageUtils.storeValuesToDoubleBuffer(storage, specData.mzs());
+      this.intensityValues = StorageUtils.storeValuesToDoubleBuffer(storage,
+          specData.intensities());
 
-  public void setDoubleBufferMzValues(double[] values) {
-    this.mzValues = StorageUtils.storeValuesToDoubleBuffer(this.storage, values);
+    } catch (MSDKException | IOException e) {
+      logger.warning("Could not load data of scan #%d".formatted(getScanNumber()));
+      return false;
+    }
+    return true;
   }
 
-  public void setDoubleBufferIntensities(double[] values) {
-    this.intensityValues = StorageUtils.storeValuesToDoubleBuffer(this.storage, values);
+  private @Nullable SimpleSpectralArrays loadData() throws MSDKException, IOException {
+    if(mzBinaryDataInfo==null) {
+      // maybe UV spectrum
+      clearUnusedData();
+      return null;
+    }
+
+    if (mzBinaryDataInfo.getArrayLength() != intensityBinaryDataInfo.getArrayLength()) {
+      logger.warning(
+          "Binary data array contains an array of different length than the default array length of the scan (#"
+          + getScanNumber() + ")");
+    }
+    double[] mzs = MzMLPeaksDecoder.decodeToDouble(mzBinaryDataInfo);
+    double[] intensities = MzMLPeaksDecoder.decodeToDouble(intensityBinaryDataInfo);
+
+    clearUnusedData();
+    return new SimpleSpectralArrays(mzs, intensities);
   }
 
-  /**
-   * Helper method used to free up the memory after memory-mapping
-   */
-  public void setTempValuesToNull() {
-    this.mzValuesTemp = null;
-    this.intensitiesTemp = null;
-    System.gc();
+  public void clearUnusedData() {
```

Comment:
we could also force garbage collection here with something like System.gc(), though not sure about that

Reply:
general idea is to never force gc.
Its done automatically when its needed